pkg/build/api/v1/types.go

```diff
@@ -171,6 +176,31 @@ type BuildSource struct {
 	SourceSecret *kapi.LocalObjectReference `json:"sourceSecret,omitempty" description:"supported auth methods are: ssh-privatekey"`
 }
 
+// ImageSource describes an image that is used as source for the build
+type ImageSource struct {
+	// From is a reference to an ImageStream tag or a DockerImage to
+	// copy source from
+	From kapi.ObjectReference `json:"from" description:"image reference"`
+
+	// Paths is a list of source and destination paths to copy from the image
+	Paths []ImageSourcePath `json:"paths" description:"paths to copy from image"`
+
+	// PullSecret is a reference to a secret to be used to pull the image from
+	// a registry
+	PullSecret *kapi.LocalObjectReference `json:"pullSecret,omitempty" description:"pull secret for source image"`
```

Comment:
We need one. Generally, we'll populate it for you if you're pointing to an image in our registry, which I'd say is the 80% case.